pkg/morph/client/notifications.go

```diff
@@ -226,6 +226,7 @@ func (c *Client) restoreSubscriptions(cli *rpcclient.WSClient, endpoint string)
 
 	// notification events restoration
 	for contract := range c.subscribedEvents {
+		contract := contract
```

Comment:
Suppose you're seeing this line a year from now, what are your immediate thoughts?

Reply:
My thoughts would be "`SubscribeForExecutionNotifications` does something asynchronously": I think this is a pretty common pattern, we even use the same name here to stress that we need a copy.

Reply:
@carpawell @acid-ant does such assignment ring any bells in you?

Reply:
Added a link on neo-go issue.

Reply:
i would understand the reason for that line ("this is a pretty common pattern" like u said) but i am always "for" any comments that describe some _really_ unexpected bugs (well, that was not bug before `neo-go` update (and i did not expect any pointer storing when wrote that code) but it is now)